model/error/event.go

```diff
@@ -147,23 +149,50 @@ func DecodeEvent(input interface{}, err error) (*Event, error) {
 	return &e, err
 }
 
-func (e *Event) Transform(config config.Config, service m.Service) common.MapStr {
+func (e *Event) Events(tctx *transform.Context) []beat.Event {
```

Comment:
An `Event e` now has a method `Events`, where the first thing it does is `transformations.Inc()` and then later on it calls `Transform()` on the event. I find that naming very confusing.

It seems to me the change from `Transform` to `Eventable` was not completely finished. 

What exactly is the logical difference now between `Transform` and `Events`?

Reply:
yeah i agree.

I needed the golang structures `transaction.Event`, `span.Span`, `metric.metric` and `error.Event` to have a method that returns `beat.Event`s. There are still `Transform` methods that have to return `common.MapStr` and I didn't want to have a situation where some `Transform` methods returned `beat.Event` and some return `common.MapStr`. So I went ahead and defined the Eventable interface for anything that has needs to create `beat.Event`'s `Events` method. I'll consider alternative naming

Reply:
I've made some changes now, let me know what you think:

* `Transform` methods now take a `transform.Context` and return `[]beat.Event`
* The common interface is called `transform.Transformable`
* `Transform` typically calls a `Fields` method on the same object which also takes a `transform.Context` but just returns the fields for the event in a common.MapStr